File: 01Singleton/Students/2017/Krzysztof Figalski/Singleton/src/simpleImplementation/Singleton.java
package simpleImplementation;

public class Singleton {

	private static Singleton instance = null;
	private static int instanceCounter = 0;

	private Singleton() {
	};

	public static Singleton getInstance() {
		if (instance == null) {
			synchronized (Singleton.class) {
				instance = new Singleton();
				instanceCounter++;
			}
		}
		return instance;
	}

	public int getInstanceCounter() {
		return instanceCounter;
	}

}

File: 01Singleton/Students/2017/LewandowskiBartosz/src/Serialization/Singleton.php
<?php

namespace Blewandowski\Serialization;


class Singleton implements \Serializable
{
    private static $instance;

    public static function getInstance()
    {
        if(!self::$instance) {
            self::$instance = new self();
        }
        return self::$instance;
    }

    public function serialize()
    {
        throw new \BadMethodCallException(
            'This class is a singleton, you can not serialize it'
        );
    }

    public function unserialize($data)
    {
        throw new \BadMethodCallException('This class is a singleton, you can not unserialize it');
    }

    public function __clone() 
    {
        throw new \BadMethodCallException('This class is a singleton, you can not clone it');
    }
}